src/core/achievements.cpp

```diff
@@ -3741,12 +3740,21 @@ static TinyString GetLoginEncryptionMachineKey()
 
   ret.resize(machine_guid_length);
   RegCloseKey(hKey);
-#elif !defined(__ANDROID__)
-#ifdef __linux__
+#else
+#if defined(__linux__)
   // use /etc/machine-id on Linux
   std::optional<std::string> machine_id = FileSystem::ReadFileToString("/etc/machine-id");
   if (machine_id.has_value())
     ret = std::string_view(machine_id.value());
+#elif defined(__APPLE__)
+  // use gethostuuid(2) on macOS
+  const struct timespec ts = { .tv_sec = 2, .tv_nsec = 0 };
+  uuid_t uuid = {};
+  if (gethostuuid(uuid, &ts) == 0)
+  {
+    for (auto c : uuid)
+      ret.append_format("{:02x}", c);
```

Comment:
```suggestion
    ret.append_hex(uuid, sizeof(uuid), false);
```

Assuming it's typedef'ed to an array.